zerver/views/auth.py

```diff
@@ -543,8 +555,73 @@ def update_login_page_context(request: HttpRequest, context: Dict[str, Any]) ->
 
     context['wrong_subdomain_error'] = WRONG_SUBDOMAIN_ERROR
 
+class PatchedExtraSessionStorage(ExtraSessionStorage):
+    def reset(self) -> None:
+        if self.prefix in self.request.session:
+            super(ExtraSessionStorage, self).reset()
+        else:
+            self.init_data()
```

Comment:
What is this patched session storage for?

Reply:
Turns out it's not even required. django-two-factor uses to store step related data in session.